flutter/lib/src/default_integrations.dart

```diff
@@ -8,6 +8,19 @@ import 'package:sentry/sentry.dart';
 import 'sentry_flutter_options.dart';
 import 'widgets_binding_observer.dart';
 
+/// It is necessary to initialize Flutter method channels so that our plugin can
+/// call into the native code.
+class WidgetsFlutterBindingIntegration
```

Comment:
should we add a test for it?
maybe taking `WidgetsFlutterBinding` as ctor param, so we can either mock it or use a fake

Reply:
The function `WidgetsFlutterBinding.ensureInitialized` is static, so we can't inject an object of `WidgetsFlutterBinding` to call/mock it.

Tried providing `WidgetsBinding Function()` as a constructor parameter with default `WidgetsFlutterBinding.ensureInitialized`. This did not work, as default parameters need to be defined at compile time.

The important part is `WidgetsFlutterBinding.ensureInitialized` beeing called in the integration, i could't figure out how to keep this behaviour and also test it. 🤔 

I did however add a test that checks if `WidgetsFlutterBindingIntegration` adds integration.

Reply:
@denrase could you do like this?

```
  WidgetsFlutterBindingIntegration({WidgetsBinding Function() ensureInitialized}) :
  _ensureInitialized = ensureInitialized ?? WidgetsFlutterBinding.ensureInitialized;

  final WidgetsBinding Function() _ensureInitialized;
```

I think it should work

Reply:
Yes, thought about that. But wouldn't we just be testing if we call the injected function, which is only there for the test in the first place? We won't be testing if `WidgetsFlutterBinding.ensureInitialized ` is called correctly.

Reply:
true, but its a unit test, and we'd like to be sure that the integration does what it's supposed to do, which is calling the `ensureInitialized()` function.